PGM/src/main/java/tc/oc/pgm/mutation/types/kit/TeamChestMutation.java

```diff
@@ -0,0 +1,135 @@
+    }
+
+    // Open shared inventory instead of placing the chest
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void onChestUse(PlayerInteractEvent event) {
+        if (event.getItem() == null) return;
+
+        if (!(event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK)) return;
+        if (event.getItem().getType() != TOOL_TYPE) return;
+
+        Player bukkitPlayer = event.getPlayer();
+        if (bukkitPlayer == null) return;
+        Optional<Inventory> oTeamInventory = getTeamsInventory(bukkitPlayer);
+
+        oTeamInventory.ifPresent(teamInventory -> {
+            event.setCancelled(true);
+            // If the item is in the off-hand slot, it wont get put back visually for the player without this.
+            if(event.getHand() == EquipmentSlot.OFF_HAND) event.getActor().updateInventory();
+            bukkitPlayer.openInventory(teamInventory);
+
+        });
+    }
+
+    @EventHandler(priority = EventPriority.LOWEST)
+    public void onInventoryClick(InventoryClickEvent event) {
+        if (event.getCurrentItem() == null) return;
+
+        // No putting evil items (ender chest, possibly wool) into the chest
+        Optional<Inventory> teamChest = getTeamsInventory(event.getActor());
+        if (teamChest.filter(teamInventory -> {
+            return teamInventory.equals(event.getView().getTopInventory());
+        }).isPresent() &&
+                isItemEvil(event.getCurrentItem())) {
+            event.setCancelled(true);
+            return;
+        }
+
+        // If normal right click, in their inventory, on the chest, then open shared inventory.
+        getTeamsInventory(event.getActor()).ifPresent(teamInventory -> {
+            if (event.getInventory().getType() == InventoryType.CRAFTING &&
+                    event.getCurrentItem().getType() == TOOL_TYPE &&
+                    event.getAction() == InventoryAction.PICKUP_HALF) {
+                event.setCancelled(true);
+                // This resets their mouse position annoyingly, but without it items can get stuck in places.
+                event.getActor().closeInventory();
+                // Prevent visual inconsistencies, specifically off-hand slot
+                if (event.getSlotType() == InventoryType.SlotType.QUICKBAR) {
+                    event.getActor().updateInventory();
+                }
+                event.getActor().openInventory(teamInventory);
+            }
+        });
+    }
+
+    private boolean isItemEvil(ItemStack item) {
+        if(item.getType() == TOOL_TYPE) return true;
+        if(oWmm.filter(wmm -> wmm.isObjectiveWool(item)).isPresent()) return true;
+
+        return false;
+    }
+
+    private Optional<Inventory> getTeamsInventory(Player bukkitPlayer) {
+        MatchPlayer player = match().getPlayer(bukkitPlayer);
+        Party team = player.getParty();
+
+        if (!team.isParticipating()) return Optional.empty();
```

Comment:
Instead of a team check, you can check the player.
`MatchPlayer player = match().participant(bukkitPlayer)`

If they aren't participating, it will be optional absent.